consensus/bor/bor.go

```diff
@@ -774,23 +783,62 @@ func (c *Bor) Seal(chain consensus.ChainReader, block *types.Block, results chan
 
 	// Wait until sealing is terminated or delay timeout.
 	log.Trace("Waiting for slot to sign and propagate", "delay", common.PrettyDuration(delay))
+	shouldSeal := make(chan bool)
+	go c.WaitForSealingOp(stop, shouldSeal, number, delay)
```

Comment:
Shouldn't we cancel the previous `SealingOp` before we wait for new? Or is it done by the `stop` channel automatically?

Reply:
Yes, the method that invokes this closes stop channel which causes the active one to stop before new sealing starts. https://github.com/maticnetwork/bor/blob/develop/miner/worker.go#L525